lib/puppet/configurer.rb

```diff
@@ -49,6 +51,14 @@ def initialize(factory = Puppet::Configurer::DownloaderFactory.new)
     @splayed = false
     @environment = Puppet[:environment]
     @transaction_uuid = SecureRandom.uuid
+    @static_catalog = true
+
+    # Get checksum types known to the agent, preferring the default specified
+    # in the digest_algorithm setting.
+    @checksum_type = valid_checksum_types
```

Comment:
The only valid values for digest_algorithm are md5 and sha256. Does it make sense to send more than that?